src/Uno.UI.RuntimeTests/Tests/HotReload/Frame/HRApp/Tests/Given_TypeReplacement.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Reflection.Metadata;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Extensions;
+using Uno.UI;
+using Uno.UI.Extensions;
+using Uno.UI.Helpers;
+using Uno.UI.RuntimeTests.Tests.HotReload.Frame.HRApp.Tests;
+using Uno.UI.RuntimeTests.Tests.HotReload.Frame.Pages;
+
+namespace Uno.UI.RuntimeTests.Tests.HotReload.Frame.HRApp.Tests;
+
+[TestClass]
+[RunsOnUIThread]
+public class Given_TypeReplacement : BaseTestClass
+{
+	public const string OriginalColor = "Green";
+	public const string UpdatedColor = "Orange";
+
+	[TestMethod]
+	public async Task When_DP_And_AttachedDP_Set_On_Old_Instance()
+	{
+		var ct = new CancellationTokenSource(TimeSpan.FromSeconds(30)).Token;
+
+		var page = new HR_DPUpdates_PageUsingComponent();
+		UnitTestsUIContentHelper.Content = page;
+
+		UserControl oldComponent = page.myComponent;
+
+		await HotReloadHelper.UpdateServerFileAndRevert<HR_DPUpdates_Component>(
+			OriginalColor,
+			UpdatedColor,
+			() =>
+			{
+				var grid = (Grid)page.Content;
+				var component = (UserControl)grid.Children.Single();
+				Assert.AreNotEqual(oldComponent, component);
+				Assert.IsTrue(component.GetType().Name.Contains("#"));
+				Assert.AreEqual(1, Grid.GetRow(component));
+				Assert.AreEqual("Hello tag", component.Tag as string);
+				// Visually, the color is updated, but this assert still fails with "Actual" being "Green" :/
+				//Assert.AreEqual(Microsoft.UI.Colors.Orange, ((SolidColorBrush)((Grid)component.Content).Background).Color);
```

Comment:
I think this we still need to fix this. @dr1rrb did you notice something similar during your tests?

Reply:
@jeromelaban This is probably an issue not relevant to the changes in this PR (i.e, an existing issues). So I think it doesn't block this PR from merge and could be looked at separately?

Reply:
The thing is that it does not validate the behavior properly, so it becomes relevant.

Reply:
Don't remember to have seen such issue in HR testing ... and I'm unable to find any good reason for this assert to fail:/

I supposed you already tried to yield the UI thread before the assertion? Otherwise in order to effectively test that changes occurred without fixing this (weird) issue (please open a new one), you can try to add a `TextBlock Text="Green"` in your component and validate that text changed ...

Reply:
> The thing is that it does not validate the behavior properly, so it becomes relevant.

I think the Tag assert and the other binding tests would fail so it still validates the fix to a good extent

Reply:
This is actually a regression, and currently there is no good way I can think of to fix this. We'll need to discuss more

Reply:
As discussed, I think we copy only the value with the `Local` precedence and only if the new instance has no value for this precedence. All other precedences are either "contextual" to where the control is being loaded or the current state of the control that should be rested by the control itself on load. The idea is only to preserve the changes made by the user since the original instance has been loaded but still applied the changes made in the xaml file.